Content.Server/Xenobiology/XenobiologySystem.cs

```diff
@@ -0,0 +1,151 @@
+using Robust.Shared.GameObjects;
+using Content.Server.Power.Components;
+using Content.Shared.Examine;
+using Robust.Shared.Localization;
+using Content.Shared.Interaction;
+using YamlDotNet.Core.Tokens;
+using Content.Shared.Tag;
+using Content.Shared.Notification.Managers;
+using Content.Server.Notification;
+using System;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Content.Shared.Audio;
+using Robust.Shared.Containers;
+
+namespace Content.Server.Xenobiology
+{
+    public class FeedEvent : EntityEventArgs
+    {
+        public IEntity? FoodItem;
+
+        public FeedEvent(IEntity item)
+        {
+            FoodItem = item;
+        }
+    }
+
+    public class XenobiologySystem : EntitySystem
+    {
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SpecimenContainmentComponent, PowerChangedEvent>(OnPowerChanged);
+            SubscribeLocalEvent<SpecimenContainmentComponent, InteractUsingEvent>(OnTubeFeed);
+            SubscribeLocalEvent<SpecimenContainmentComponent, ExaminedEvent>(OnContainmentExamined);
+            SubscribeLocalEvent<SpecimenDietComponent, ExaminedEvent>(OnSpecimenExamined);
+            SubscribeLocalEvent<SpecimenDietComponent, FeedEvent>(OnSpecimenFed);
+        }
+
+       
+
+        /// <summary>
+        /// Builds a dynamic specimen examination
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnSpecimenExamined(EntityUid uid, SpecimenDietComponent component, ExaminedEvent args)
+        {
+            args.Message.AddText("\n");
+            args.Message.AddMarkup(Loc.GetString("specimen-hungry") + (" "));
+            args.Message.AddMarkup($"[color=#f6ff05]{component.SelectedDiet}[/color]");
+        }
+
+        /// <summary>
+        /// Occurs when you feed the specimen a specific item
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="comp"></param>
+        /// <param name="args"></param>
+        public void OnSpecimenFed(EntityUid uid, SpecimenDietComponent comp, FeedEvent args)
+        {
+           var specimen = comp.Owner;
+           var usedItem = args.FoodItem;
+           //If the usedItem has a certian tag, which corresponds to the picked diet, we increase the growth state
+           //Otherwise, output a message about the wrong food choice
+           if (usedItem == null) return;
+           if (usedItem.TryGetComponent<TagComponent>(out TagComponent? ItemTag))
+           {
+                if (ItemTag.Tags.Contains(comp.SelectedDiet))
+                {
+                    specimen.PopupMessageEveryone(Loc.GetString("specimen-humanlike-fed"));
+                    EntityManager.QueueDeleteEntity(usedItem.Uid); //Delete the food
```

Comment:
Just using tag is kinda bad because you can just split food to min/max

Reply:
Since it chooses a new food item each time you can't

Reply:
Except you can? It doesn't check for any solutions.